plugin/pkg/admission/priority/admission.go

```diff
@@ -192,11 +188,15 @@ func (p *priorityPlugin) validatePriorityClass(a admission.Attributes) error {
 	if !ok {
 		return errors.NewBadRequest("resource was marked with kind PriorityClass but was unable to be converted")
 	}
-	if pc.Value > schedulerapi.HighestUserDefinablePriority {
-		return admission.NewForbidden(a, fmt.Errorf("maximum allowed value of a user defined priority is %v", schedulerapi.HighestUserDefinablePriority))
-	}
-	if _, ok := schedulerapi.SystemPriorityClasses[pc.Name]; ok {
-		return admission.NewForbidden(a, fmt.Errorf("the name of the priority class is a reserved name for system use only: %v", pc.Name))
+	// API server adds system critical priority classes at bootstrapping. We should
+	// not enforce restrictions on adding system level priority classes for API server.
+	if userInfo := a.GetUserInfo(); userInfo == nil || userInfo.GetName() != user.APIServerUser {
+		if pc.Value > scheduling.HighestUserDefinablePriority {
+			return admission.NewForbidden(a, fmt.Errorf("maximum allowed value of a user defined priority is %v", scheduling.HighestUserDefinablePriority))
+		}
+		if strings.HasPrefix(pc.Name, scheduling.SystemPriorityClassPrefix) {
+			return admission.NewForbidden(a, fmt.Errorf("priority class names with '"+scheduling.SystemPriorityClassPrefix+"' prefix are reserved for system use only"))
+		}
 	}
 	// If the new PriorityClass tries to be the default priority, make sure that no other priority class is marked as default.
 	if pc.GlobalDefault {
```

Comment:
unrelated, but it still appears we're trying to enforce a singleton GlobalDefault... I thought we dropped that

Reply:
We never dropped it. We just added code to handle the case of having more than one global default when they are added due to race in HA clusters.

Reply:
ah, that probably means admission can reject updates that swap which one is the default

1. apply a file containing two priority classes, `a` and `b` (default)
2. apply an update of the file with `a` (default) and `b` (non-default)

admission rejects if the update to `a` happens first. it would also reject if the update to `b` happened first but the watch event didn't make it into the informer-fed cache the admission plugin consulted.

this guard seems more problematic than helpful to me

Reply:
I agree that there are races causing one to add more than one global default, but are you suggesting that it is better to allow people to add as many global default priority classes as they like?

Reply:
if we cannot guarantee correctness (single default), and we have a deterministic and documented behavior when there's more than one (which we do), then breaking legitimate updates in order to half-accomplish correctness doesn't seem worthwhile

Reply:
I see your point. We can remove it and document the behavior. I will do that in a separate PR to ensure we are not mixing the two.